api/apiclient.go

```diff
@@ -657,8 +657,25 @@ func gorillaDialWebsocket(ctx context.Context, urlStr string, tlsConfig *tls.Con
 // specified addresses will be attempted concurrently, and the first
 // successful connection will be returned.
 func dialWebsocketMulti(ctx context.Context, addrs []string, path string, opts dialOpts) (*dialResult, error) {
+	// Prioritise non-dial errors over the normal "connection refused".
+	isDialError := func(err error) bool {
+		netErr, ok := errors.Cause(err).(*net.OpError)
+		if !ok {
+			return false
+		}
+		return netErr.Op == "dial"
+	}
+	combine := func(initial, other error) error {
+		if initial == nil || isDialError(initial) {
```

Comment:
Do you need `initial == nil` here? It seems like isDialError would return false for a nil, right? If not, might be cleaner to change it so that it does.

Reply:
If we don't have the nil check, we would never return a dial error even if it was the only error.